EFWithRepoPattern/Data/Repositories/Core/IRepository.cs

```diff
@@ -3,7 +3,7 @@
 namespace EFWithRepoPattern.Data.Repositories.Core
 {
     /// <summary>
-    /// Provides basic functionality for pulling and pushing data to a database related to the given
+    /// Provides basic functionality for pulling and pushing data to a datasource related to the given
```

Comment:
I do like this way of expressing that the interface does!!  But I prefer the suggestion below.

```suggestion
    /// Provides basic functionality for pulling and pushing data to a datasource/database related to the given
```